File: ZKCache/src/main/java/com/ericsson/ema/tim/dml/Selector.java
package com.ericsson.ema.tim.dml;

import com.ericsson.ema.tim.dml.predicate.Predicate;

import java.util.List;

/**
 * Created by eqinson on 2017/4/18.
 */
public interface Selector {
    Selector from(String tab);

    Selector where(Predicate predicate);

    Selector orderby(String field, String asc);

    Selector orderby(String field);

    Selector limit(int limit);

    Selector skip(int skip);

    List<Object> execute();

    List<List<Object>> executeWithSelectFields();
}

File: zkcache_scala/src/main/scala/com/ericsson/ema/tim/dml/Selector.scala
package com.ericsson.ema.tim.dml

import com.ericsson.ema.tim.dml.predicate.Predicate

/**
  * Created by eqinson on 2017/5/12.
  */
trait Selector {
	def from(tab: String): Selector

	def where(predicate: Predicate): Selector

	def limit(limit: Int): Selector

	def skip(skip: Int): Selector

	def collect(): List[Object]

	def collectBySelectFields(): List[List[Object]]

	def count(): Long

	def exists(): Boolean
}
